### PR TITLE
 ✨Only update lists when there was an update

### LINKED ISSUE #872: Only Update Lists when there was an update
#### Describe your issue

Right now all lists in BPMN-Studio (Solution Explorer in PE View for example) get refreshed everytime the component polls for new diagrams even though most of the time no update should happen since there is no new content to render.

#### How can others reproduce the issue?

Start BPMN-Studio.
Check for the refreshing of lists.

#### Possible solution

Check if there was an update to the content of the list, if not don't update the corresponding array.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by Vyperus**
:tada: **Thanks for submitting this Pull Request**

Please provide a list including your changes.

To help us to create a meaningful Changelog, we agreed 
to use an emoji-based commit-style. Use the [Emojis](#emojis)
we provided below.

Fill out the following, provide reference to issues
and this PR- we will use this as merge commit message-body
and the title of this PR as commit-headline:

---

**Changes:**

1. Adds simple if-case to only update the displayed list when there was an update to this list.

**Issues:**

Closes #872 

PR: #PullRequest

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/process-solution-panel/process-solution-panel.ts [hunk 1]
```diff
@@ -563,8 +563,10 @@ export class ProcessSolutionPanel {
 
     try {
       await this._solutionExplorerServiceManagementApi.openSolution(processengineSolutionString, this._solutionExplorerIdentity);
-      this.openedProcessEngineSolution = await this._solutionExplorerServiceManagementApi.loadSolution();
-
+      const openedSolution: ISolution = await this._solutionExplorerServiceManagementApi.loadSolution();
+      if (JSON.stringify(openedSolution) !== JSON.stringify(this.openedProcessEngineSolution)) {
```
Comment (robpal94) [inline src/modules/process-solution-panel/process-solution-panel.ts]: can we use a dedicated variable for this?

Comment (SteffenKn) [reply src/modules/process-solution-panel/process-solution-panel.ts]: Of course. Always move the business rule into an own variable.

Comment (SteffenKn) [inline src/modules/process-solution-panel/process-solution-panel.ts]: Also: Add some spacing
```
      const openedSolution: ISolution = await this._solutionExplorerServiceManagementApi.loadSolution();

      if (JSON.stringify(openedSolution) !== JSON.stringify(this.openedProcessEngineSolution)) {
```

File: src/modules/processdef-list/processdef-list.ts [hunk 2]
```diff
@@ -104,7 +104,13 @@ export class ProcessDefList {
 
   private async _getAllProcessModels(): Promise<void> {
     const processModelExecution: ProcessModelExecution.ProcessModelList = await this._managementApiClient.getProcessModels(this._managementContext);
-    this.allProcessModels = processModelExecution.processModels;
+
+    const listWasUpdatet: boolean = JSON.stringify(processModelExecution.processModels) !== JSON.stringify(this.allProcessModels);
```
Comment (SteffenKn) [inline src/modules/processdef-list/processdef-list.ts]: listWasUpdate**d**

File: src/modules/dashboard/dashboard.ts [hunk 3]
```diff
@@ -213,7 +213,13 @@ export class Dashboard {
 
   private async _updateUserTasks(): Promise<void> {
     try {
-      this._userTasks = await this._getUserTasks();
+      const userTasks: Array<IUserTaskWithProcessModel> = await this._getUserTasks();
+      const userTaskListWasUpdatet: boolean = JSON.stringify(userTasks) !== JSON.stringify(this._userTasks);
```
Comment (SteffenKn) [inline src/modules/dashboard/dashboard.ts]: userTaskListWasUpdate**d**

File: src/modules/dashboard/dashboard.ts [hunk 4]
```diff
@@ -213,7 +213,13 @@ export class Dashboard {
...
+
+      if (userTaskListWasUpdatet) {
```
Comment (SteffenKn) [inline src/modules/dashboard/dashboard.ts]: userTaskListWasUpdate**d**

File: src/modules/process-list/process-list.ts [hunk 5]
```diff
@@ -67,7 +67,13 @@ export class ProcessList {
 
   public async updateProcesses(): Promise<void> {
     try {
-      this._processes = await this._getProcesses();
+      const processes: Array<Correlation> = await this._getProcesses();
+      const processListWasUpdatet: boolean = JSON.stringify(processes) !== JSON.stringify(this._getProcesses);
```
Comment (SteffenKn) [inline src/modules/process-list/process-list.ts]: processListWasUpdate**d**

File: src/modules/process-list/process-list.ts [hunk 6]
```diff
@@ -67,7 +67,13 @@ export class ProcessList {
...
+
+      if (processListWasUpdatet) {
```
Comment (SteffenKn) [inline src/modules/process-list/process-list.ts]: processListWasUpdate**d**

File: src/modules/process-solution-panel/process-solution-panel.ts [hunk 7]
```diff
@@ -563,7 +563,12 @@ export class ProcessSolutionPanel {
 
     try {
       await this._solutionExplorerServiceManagementApi.openSolution(processengineSolutionString, this._solutionExplorerIdentity);
-      this.openedProcessEngineSolution = await this._solutionExplorerServiceManagementApi.loadSolution();
+      const openedSolution: ISolution = await this._solutionExplorerServiceManagementApi.loadSolution();
+      const solutionWasUpdatet: boolean = JSON.stringify(openedSolution) !== JSON.stringify(this.openedProcessEngineSolution);
```
Comment (SteffenKn) [inline src/modules/process-solution-panel/process-solution-panel.ts]: solutionWasUpdate**d**

File: src/modules/process-solution-panel/process-solution-panel.ts [hunk 8]
```diff
@@ -563,7 +563,12 @@ export class ProcessSolutionPanel {
...
+
+      if (solutionWasUpdatet) {
```
Comment (SteffenKn) [inline src/modules/process-solution-panel/process-solution-panel.ts]: solutionWasUpdate**d**

File: src/modules/processdef-list/processdef-list.ts [hunk 9]
```diff
@@ -104,7 +104,13 @@ export class ProcessDefList {
...
+
+    if (listWasUpdatet) {
```
Comment (SteffenKn) [inline src/modules/processdef-list/processdef-list.ts]: listWasUpdate**d**

File: src/modules/task-list/task-list.ts [hunk 10]
```diff
@@ -63,7 +63,13 @@ export class TaskList {
 
   private async updateUserTasks(): Promise<void> {
     try {
-      this._userTasks = await this._getUserTasks();
+      const userTasks: Array<IUserTaskWithProcessModel> = await this._getUserTasks();
+      const userTaskListWasUpdatet: boolean = JSON.stringify(userTasks) !== JSON.stringify(this._userTasks);
```
Comment (SteffenKn) [inline src/modules/task-list/task-list.ts]: userTaskListWasUpdate**d**

File: src/modules/task-list/task-list.ts [hunk 11]
```diff
@@ -63,7 +63,13 @@ export class TaskList {
...
+
+      if (userTaskListWasUpdatet) {
```
Comment (SteffenKn) [inline src/modules/task-list/task-list.ts]: userTaskListWasUpdate**d**

**Pull request merged**